raiden/network/transport/matrix/client.py

```diff
@@ -695,15 +697,18 @@ def _handle_responses(
 
                 room = self.rooms[room_id]
                 room.prev_batch = sync_room["timeline"]["prev_batch"]
+                room_members_count = len(room._members)
 
                 for event in sync_room["state"]["events"]:
                     event["room_id"] = room_id
                     room._process_state_event(event)
-
                 for event in sync_room["timeline"]["events"]:
                     event["room_id"] = room_id
                     room._put_event(event)
 
+                # number of members changed. Verify validity of room
+                if room_members_count != len(room._members):
```

Comment:
Why do we need this check? I would assume that a "join" event only happens when the number of members changes.

Reply:
In theory there could be a leave event right after a join event. I don't think it makes sense to keep track of the state manually here.